library/pkcs7.c

```diff
@@ -664,6 +788,30 @@ static int mbedtls_pkcs7_data_or_hash_verify(mbedtls_pkcs7 *pkcs7,
             continue;
         }
 
+        if (signer->signed_attrs_raw.len > 0)
+        {
+            ret = pkcs7_signed_attrs_match(&signer->signed_attrs, hash,
+                                           mbedtls_md_get_size(md_info));
+            if (ret != 0) {
+                mbedtls_free(hash);
+                continue;
+            }
+
+            mbedtls_md_init(&md_ctx);
+            if (mbedtls_md_setup(&md_ctx, md_info, 0) != 0 ||
+                mbedtls_md_starts(&md_ctx) != 0 ||
+                mbedtls_md_update(&md_ctx, &(unsigned char){0x31}, 1) != 0 ||
+                mbedtls_md_update(&md_ctx, signer->signed_attrs_raw.p + 1,
+                                  signer->signed_attrs_raw.len - 1) != 0 ||
+                mbedtls_md_finish(&md_ctx, hash) != 0) {
+                mbedtls_md_free(&md_ctx);
+                mbedtls_free(hash);
+                ret = MBEDTLS_ERR_PKCS7_VERIFY_FAIL;
+                continue;
+            }
+            mbedtls_md_free(&md_ctx);
```

Comment:
These errors indicate “something unexpected went wrong”.  `MBEDTLS_ERR_PKCS7_ALLOC_FAILED` seems like a more plausible error, though the specific error would be even better.